packages/tools/fluid-runner/src/exportFile.ts

```diff
@@ -0,0 +1,115 @@
+    if (fs.existsSync(telemetryFile)) {
+        return {
+            success: false,
+            eventName: clientArgsValidationError,
+            errorMessage: `Telemetry file already exists [${telemetryFile}]`,
+        };
+    }
+
+    const fileLogger = new FileLogger(telemetryFile, 50);
+
+    const logger = ChildLogger.create(fileLogger, "LocalSnapshotRunnerApp",
+        { all: { Event_Time: () => Date.now() } });
+
+    try {
+        return await PerformanceEvent.timedExecAsync(logger, { eventName: "ExportFile" }, async () => {
+            // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
+            const codeLoaderBundle = require(codeLoader);
+            if (!isCodeLoaderBundle(codeLoaderBundle)) {
+                const eventName = clientArgsValidationError;
+                const message = "Code loader bundle is not of type CodeLoaderBundle";
+                logger.sendErrorEvent({
+                    eventName,
+                    message,
+                });
+                return { success: false, eventName, errorMessage: message };
+            }
+
+            const argsValidationError = getArgsValidationError(inputFile, outputFolder, scenario);
+            if (argsValidationError) {
+                const eventName = clientArgsValidationError;
+                logger.sendErrorEvent({
+                    eventName,
+                    message: argsValidationError,
+                });
+                return { success: false, eventName, errorMessage: argsValidationError };
+            }
+
+            // TODO: read file stream
+            const inputFileContent = fs.readFileSync(inputFile, { encoding: "utf-8" });
+
+            const results = await createContainerAndExecute(
+                inputFileContent,
+                logger,
+                await codeLoaderBundle.fluidExport,
+            );
+            // eslint-disable-next-line guard-for-in, no-restricted-syntax
+            for (const key in results) {
+                fs.appendFileSync(path.join(outputFolder, key), results[key]);
+            }
+            return { success: true };
+        });
+    } catch (error) {
+        const eventName = "Client_UnexpectedError";
+        logger.sendErrorEvent({ eventName }, error);
+        return { success: false, eventName, errorMessage: "Unexpected error" };
+    } finally {
+        await fileLogger.flush();
+    }
+}
+
+export async function createContainerAndExecute(
+    localOdspSnapshot: string,
+    logger: ITelemetryLogger,
+    fluidFileConverter: IFluidFileConverter,
+): Promise<Record<string, string>> {
+    const loader = new Loader({
+        urlResolver: new FakeUrlResolver(),
+        documentServiceFactory: createLocalOdspDocumentServiceFactory(localOdspSnapshot),
+        codeLoader: fluidFileConverter.codeLoader,
+        scope: fluidFileConverter.scope,
+    });
+
+    const container = await loader.resolve({ url: "/fakeUrl/" });
```

Comment:
We need to have IContainerLoadOptions.loadMode.opsBeforeReturn: "cached" to process any trailing ops in snapshot.
It looks like it's passed through     [LoaderHeader.loadMode]: IContainerLoadMode; header

Reply:
And deltaConnection: "none" would be nice (but I think not required - due to usage of storageOnly mode)